### PR TITLE
Pooltype interface

### PR DESCRIPTION
**Pull request by fluidvanadium**
allows improvement for a test

### CODE REVIEW

File: zingolib/src/wallet/notes/transparent.rs [hunk 1]
```diff
@@ -53,6 +54,47 @@ impl OutputInterface for TransparentOutput {
     fn pending_spent_mut(&mut self) -> &mut Option<(TxId, u32)> {
         &mut self.unconfirmed_spent
     }
+
+    fn transaction_record_to_outputs_vec(transaction_record: &TransactionRecord) -> Vec<&Self> {
+        let mut vecref = vec![];
+        for output in &transaction_record.transparent_outputs {
+            vecref.push(output);
+        }
+        vecref
```
Comment (AloeareV) [inline zingolib/src/wallet/notes/transparent.rs]: Fluent is simpler here
```suggestion
        transaction_record.sapling_notes.iter().collect()
```

File: zingolib/src/wallet/notes/transparent.rs [hunk 2]
```diff
@@ -53,6 +54,47 @@ impl OutputInterface for TransparentOutput {
...
+        transaction_record: &TransactionRecord,
+        spend_status_query: OutputSpendStatusQuery,
+    ) -> Vec<&Self> {
+        let mut vecref = vec![];
+        for output in &transaction_record.transparent_outputs {
+            if output.spend_status_query(spend_status_query) {
+                vecref.push(output);
+            }
+        }
+        vecref
```
Comment (AloeareV) [inline zingolib/src/wallet/notes/transparent.rs]: Not as many fewer lines but still likely easier to follow
```suggestion
        transaction_record
            .sapling_notes
            .iter()
            .filter(|output| output.spend_status_query(spend_status_query))
            .collect()

```

File: zingolib/src/wallet/notes/transparent.rs [hunk 3]
```diff
@@ -53,6 +54,47 @@ impl OutputInterface for TransparentOutput {
...
+        spend_status_query: OutputSpendStatusQuery,
+    ) -> Vec<&mut Self> {
+        let mut vecrefmut = vec![];
+        for output in &mut transaction_record.transparent_outputs {
+            if output.spend_status_query(spend_status_query) {
+                vecrefmut.push(output);
+            }
+        }
+        vecrefmut
+    }
```
Comment (AloeareV) [inline zingolib/src/wallet/notes/transparent.rs]: As above but with iter_mut().

**Review by AloeareV — CHANGES_REQUESTED** [review]
I think some iterators would be simpler than these for loops

File: zingolib/src/wallet/notes/transparent.rs (hunk 1 above)
Comment (fluidvanadium) [reply zingolib/src/wallet/notes/transparent.rs]: AWESOME

File: zingolib/src/wallet/notes/transparent.rs (hunk 3 above)
Comment (fluidvanadium) [reply zingolib/src/wallet/notes/transparent.rs]: im doing it. thank you for the pointer!

**Review by AloeareV — APPROVED** [review]

**Pull request merged**